dvc/repo/experiments/utils.py

```diff
@@ -203,3 +203,59 @@ def check_ref_format(scm: "Git", ref: ExpRefInfo):
             f"Invalid exp name {ref.name}, the exp name must follow rules in "
             "https://git-scm.com/docs/git-check-ref-format"
         )
+
+
+def name2exp_ref(
```

Comment:
If you look at the implementation for `resolve_exp_ref`, it will end up iterating over all local or remote names (when we use functions like `exp_refs_by_name` around like 165). So this `name2exp_ref` implementation will make us do that full list fetch/iteration for each name in `exp_names`.

I think this whole workflow needs to be refactored so that we only have to iterate over the full list of local or remote experiments once:

- the `(remote)_exp_refs_by_name` methods should accept multiple names, and return a dict mapping a name we searched for to the list of matches for that name.
- `resolve_exp_ref` would also accept multiple names, and raise an exception as soon as one of them could not be resolved (i.e. as soon as we get a dict item with `len(values) != 1`)